core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -0,0 +1,112 @@
+KOKKOS_INLINE_FUNCTION auto array_layout_from_mapping(
+    const typename MDSpanType::mapping_type &mapping) {
+  using mapping_type = typename MDSpanType::mapping_type;
+  using extents_type = typename mapping_type::extents_type;
+
+  constexpr auto rank = extents_type::rank();
+  const auto &ext     = mapping.extents();
+
+  static_assert(rank <= ARRAY_LAYOUT_MAX_RANK,
+                "Unsupported rank for mdspan (must be <= 8)");
+
+  if constexpr (std::is_same_v<ArrayLayout, LayoutStride>) {
+    return Kokkos::LayoutStride{
+        rank > 0 ? ext.extent(0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 0 ? mapping.stride(0) : 0,
+        rank > 1 ? ext.extent(1) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 1 ? mapping.stride(1) : 0,
+        rank > 2 ? ext.extent(2) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 2 ? mapping.stride(2) : 0,
+        rank > 3 ? ext.extent(3) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 3 ? mapping.stride(3) : 0,
+        rank > 4 ? ext.extent(4) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 4 ? mapping.stride(4) : 0,
+        rank > 5 ? ext.extent(5) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 5 ? mapping.stride(5) : 0,
+        rank > 6 ? ext.extent(6) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 6 ? mapping.stride(6) : 0,
+        rank > 7 ? ext.extent(7) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 7 ? mapping.stride(7) : 0,
+    };
+  } else {
+    return ArrayLayout{rank > 0 ? ext.extent(0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 1 ? ext.extent(1) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 2 ? ext.extent(2) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 3 ? ext.extent(3) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 4 ? ext.extent(4) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 5 ? ext.extent(5) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 6 ? ext.extent(6) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+                       rank > 7 ? ext.extent(7) : KOKKOS_IMPL_CTOR_DEFAULT_ARG};
+  }
+}
+
+template <class MDSpanType, class VM>
+KOKKOS_INLINE_FUNCTION auto mapping_from_view_mapping(const VM &view_mapping) {
+  using mapping_type = typename MDSpanType::mapping_type;
+  using extents_type = typename mapping_type::extents_type;
+
+  if constexpr (std::is_same_v<typename mapping_type::layout_type,
+                               Kokkos::layout_stride>) {
+    // std::span is not available in C++17 (our current requirements),
+    // so we need to use the std::array constructor for layout mappings.
+    // When C++20 is available, we can use std::span here instead
+    std::array<std::size_t, VM::Rank> strides;
+    view_mapping.stride_fill(strides.data());
+    return mapping_type(extents_from_view_mapping<extents_type>(view_mapping),
+                        strides);
+  } else {
+    return mapping_type(extents_from_view_mapping<extents_type>(view_mapping));
```

Comment:
we are missing the stride of LayoutLeft/Right if we construct padded layout

Reply:
I'd be ok adding assert that the stride equals the extent in debug mode for now and then fix it in a subsequent PR with added test. This is likely not fully trivial.

Reply:
I added a test for padded layouts. Basically if the rank is > 1 I initialize the argument with the actual stride from the view now.